jerry-util/arg-validator/arg-validator.c

```diff
@@ -0,0 +1,98 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "arg-validator.h"
+#include "jerryscript.h"
+
+/**
+ * It validate the JS arguments and assign them to the native argument.
+ *
+ * @return jerry undefined: the validator passes,
+ *         jerry error: the validator fails.
+ */
+jerry_value_t
+jerry_validate_and_assign_args (const jerry_value_t *js_arg_p, /**< points to the js argument */
+                                const jerry_length_t js_arg_cnt, /**< the count of the js arguments */
+                                const jerry_arg_t *c_arg_p, /**< points to the jerry_arg_t */
+                                const jerry_length_t c_arg_cnt) /**< the count of the native argument */
+{
+  jerry_value_t ret = jerry_create_undefined ();
+
+  if (c_arg_cnt == 0)
+  {
+    return ret;
+  }
+
+  jerry_arg_iterator_t iterator =
+  {
+    .c_arg_p = c_arg_p,
+    .js_arg_p = js_arg_p,
+    .js_arg_cnt = js_arg_cnt,
+    .js_arg_index = 0
+  };
+
+  jerry_length_t c_arg_index = 0;
+
+
+  while (c_arg_index < c_arg_cnt)
+  {
+    iterator.c_arg_p = &c_arg_p[c_arg_index];
+
+    ret = c_arg_p[c_arg_index].func (&iterator);
+
+    if (jerry_value_has_error_flag (ret))
+    {
+      break;
+    }
+
+    c_arg_index ++;
+  }
+
+  return ret;
+} /* jerry_validate_and_assign_args */
+
+/**
+ * It validate the JS arguments and this value,
+ * and assign them to the native argument.
+ *
+ * @return jerry undefined: the validator passes,
+ *         jerry error: the validator fails.
+ */
+jerry_value_t
+jerry_validate_and_assign_all (const jerry_value_t this_val, /**< the this_val for the external function */
+                               const jerry_value_t *js_arg_p, /**< points to the js argument */
+                               const jerry_length_t js_arg_cnt, /**< the count of the js arguments */
+                               const jerry_arg_t *c_arg_p, /**< points to the jerry_arg_t */
+                               const jerry_length_t c_arg_cnt) /**< the count of the native argument */
+{
+  jerry_value_t ret = jerry_create_undefined ();
+
+  if (c_arg_cnt == 0)
+  {
+    return ret;
```

Comment:
Does it make sense to `return jerry_create_undefined()` here and not initialize `ret` in line 80?